lib/ramble/ramble/pipeline.py

```diff
@@ -319,13 +319,33 @@ def _complete(self):
                  | stat.S_IROTH | stat.S_IXOTH)
 
 
-pipelines = Enum('pipelines', ['analyze', 'archive', 'mirror', 'setup'])
+class PushToCachePipeline(Pipeline):
+    """Class for the pushtocache pipeline"""
+
+    name = 'pushtocache'
+
+    def __init__(self, workspace, filters, spack_cache_path=None):
+        super().__init__(workspace, filters)
+        self.action_string = 'Pushing to Spack Cache'
+        self.spack_cache_path = spack_cache_path
+
+    def _prepare(self):
+        super()._prepare()
+        self.workspace.spack_cache_path = self.spack_cache_path
+
+    def _complete(self):
+        tty.msg('Pushed envs to spack cache %s' % self.spack_cache_path)
+
+
+# TODO: should this use the classes `.name`?
```

Comment:
Yeah, I think there are several ways we can remove this. I think I only left it for error checking (i.e. trying to create a pipeline that doesn't exist yet)

Reply:
OK I will make a task to clean this up and remove the TODO here